web/server/routers/account.ts

```diff
@@ -85,4 +90,50 @@ export const accountRouter = router({
         });
       }
     }),
+
+  add: protectedProcedure
+    .input(
+      z.object({
+        email: z.string().email(),
+        admin: z.boolean(),
+      })
+    )
+    .mutation(async ({ ctx, input }) => {
+      if (ctx.session?.email === input.email) {
+        throw new TRPCError({
+          code: "UNAUTHORIZED",
+          message: "User cannot add themselves",
+        });
+      }
+
+      if ((await findAccount(input.email)).length != 0) {
+        throw new TRPCError({
+          code: "UNAUTHORIZED",
+          message: "Account already exists",
+        });
+      }
+
+      const session = await Account.startSession();
+      session.startTransaction();
+
+      try {
+        const inputData: [string, boolean] = [input.email, input.admin];
+        await addAccount(inputData, session);
+        if ((await findUser(input.email)).length != 0) {
+          await updateUser(
+            input.email,
+            { admin: input.admin, disabled: false },
+            session
+          );
+        }
+        session.commitTransaction();
+        return { success: true };
+      } catch (e) {
+        session.abortTransaction();
+        throw new TRPCError({
+          code: "INTERNAL_SERVER_ERROR",
+          message: "An unexpected error occurred",
+        });
+      }
+    }),
```

Comment:
You can minimize the db calls by removing `findAccount()` - `Model.create` will return an error if you try to make a document with an existing email since it's a unique key so you can remove the additional call to check if the account exists in the db. Similarly, `findOneAndUpdate` also only updates a document if a document matching the query is found.